### ISSUE TITLE
[Feature request] Support search by trigger like `@@` and enter to link to the note

### ISSUE THREAD

**Issue by Quorafind** [scambier/obsidian-omnisearch#32]
**Is your feature request related to a problem? Please describe.**

When I want to refer note, I need to 

1. search its name, 
2. and then remember its name, 
3. finally input the name into my note to link it.

**Describe the solution you'd like**

If I can search the content by use trigger like `[[` or `@@` in obsidian and then enter to link it, it will be super easy for me to link my note. 
![image](https://user-images.githubusercontent.com/13215013/165661424-2d389f00-ce69-4a18-a2fb-b1e57ac3d004.png)

**Describe alternatives you've considered**

If I can search note and then quick copy the name.

**Additional context**

**Comment by dartungar** [comment]
Another alternatve would be a hotkey (like Alt+Enter) that, when pressed from Omnisearch modal, pastes currently selected (in Omnisearch modal) note as a [[link]] at current cursor position in active note.

**Comment by gobbletown** [comment]
This feature would make the Omnisearch plugin the most useful Community plugin created yet!

Full-text search with a single "paste" function from found search results to current active note cursor, with a link, embed, or block-ref would be game-changing for facilitating inline, realtime, robust inter-linkages within Obsidian!

Like so useful and unmatched in simplicity or elegance with any other in the Obsidian ecosystem.

@scambier @Quorafind Any update or plans on implementation for this?

**Comment by scambier** [comment]
@gobbletown no real plans, besides what I'm doing for the [next release](https://github.com/scambier/obsidian-omnisearch/milestone/1). This feature shouldn't be too hard to implement, I'll certainly work on it for 1.4

**Comment by gobbletown** [comment]
much obliged!  can't wait to see 1.4!

aliasing tweak you have planned for 1.3 will be great as well.

thanks!

**Comment by scambier** [comment]
Ok so I tried implementing a feature like `[@<search>]`, but I was not satisfied with the usage (it's slow to write and feeds like a hidden feature), and it would have required additional work to correctly style the results list.

Instead, the `alt+enter` shortcut (previously "expand in-note results") has been reassigned to "insert a link". I assigned `tab` to "switch to in-file/vault search", which IMO also feels easier to use.
![image](https://user-images.githubusercontent.com/3216752/175768927-393c0518-44c4-4a69-8b54-611f86052add.png)
